agent/mapcall/src/main/java/randoop/instrument/MapCallsAgent.java

```diff
@@ -1,322 +1,195 @@
 package randoop.instrument;
 
-import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileReader;
 import java.io.IOException;
 import java.io.InputStream;
-import java.lang.instrument.ClassFileTransformer;
+import java.io.InputStreamReader;
+import java.io.Reader;
 import java.lang.instrument.Instrumentation;
-import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Enumeration;
-import java.util.HashSet;
 import java.util.LinkedHashSet;
 import java.util.List;
-import java.util.jar.JarEntry;
-import java.util.jar.JarFile;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import plume.EntryReader;
 import plume.Option;
 import plume.Options;
-import plume.SimpleLog;
-
+import randoop.MethodReplacements;
+
+/**
+ * The MapCalls javaagent applies the {@link CallReplacementTransformer} to map calls in loaded
+ * classes to alternate implementations.
+ *
+ * <p>The transformer applies method call replacements as specified in either the default or a user
+ * provided replacement file. (See the <a
+ * href="https://randoop.github.io/randoop/manual/index.html#map_calls">mapcall user
+ * documentation</a> the file format.) Default replacements are given in an internal resource file
+ * {@code "default-replacements.txt"}. User replacements are then loaded using the {@link #map_calls
+ * --map-calls} command-line argument. A user replacement may override a default replacement.
+ *
+ * <p>The classes of packages listed in the resource file {@code "default-load-exclusions.txt"} are
+ * excluded from transformation.
+ */
 public class MapCallsAgent {
 
+  /** Run the mapcall agent in debug mode. */
+  @SuppressWarnings("WeakerAccess")
   @Option("print debug information")
   public static boolean debug = false;
 
+  @SuppressWarnings("WeakerAccess")
+  @Option("directory name for debug information")
+  public static String debug_directory = "";
+
+  static Path debugPath = Paths.get("").toAbsolutePath().normalize();
```

Comment:
It's confusing to set a concrete value that is intended to be changed later.  This doesn't need to be initialized at all.  It can be set at the location where it is currently re-set, and then all the logic about how it is set is in one place.  Right now, looking either here or there doesn't give the full story.